api/location/schema.py

```diff
@@ -30,6 +31,33 @@ def mutate(self, info, **kwargs):
         return CreateLocation(location=location)
 
 
+class UpdateLocation(graphene.Mutation):
+    class Arguments:
+        location_id = graphene.Int()
+        name = graphene.String()
+        abbreviation = graphene.String()
+        country = graphene.String()
+        image_url = graphene.String()
+        time_zone = graphene.String()
+    location = graphene.Field(Location)
+
+    def mutate(self, info, location_id, **kwargs):
+        location = Location.get_query(info)
+        location_object = location.filter(
+            LocationModel.id == location_id).first()
+        if not location_object:
+            raise GraphQLError("Location not found")
+        if "time_zone" in kwargs:
+            validate_timezone_field(**kwargs)
+        if "country" in kwargs:
+            validate_country_field(**kwargs)
+        if "name" in kwargs:
```

Comment:
You might need to check if all fields beside the fields already validated are available and validate if they are empty, this would only validate for the name field, remember this leaves, imageURL, abbreviation and timezone fields un validate

Reply:
time_zone is already taken care of using `validate_timezone_field`, so I will consider validating empty abbreviation and image URL  together with the name.